File: src/main/java/com/example/travelator/optional/java/Leg.java
package com.example.travelator.optional.java;

import java.time.Duration;
import java.util.Optional;

public class Leg {
	private final String name;
	private final Duration plannedDuration;
	
	private Leg(String name, Duration plannedDuration) {
		this.name = name;
		this.plannedDuration = plannedDuration;
	}
	
	public static Leg leg(String name, Duration duration) {
		return new Leg(name, duration);
	}
	
	public Duration getPlannedDuration() {
		return plannedDuration;
	}
	
	public String getName() {
		return name;
	}
	
	public String getDescription() {
		return name;
	}
}

File: src/main/java/com/example/travelator/optional/kotlin/Leg.kt
package com.example.travelator.optional.kotlin

import kotlin.time.Duration

class Leg private constructor(val description: String, val plannedDuration: Duration) {


    companion object {
        @JvmStatic
        fun leg(name: String, duration: Duration): Leg {
            return Leg(name, duration)
        }
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Leg) return false

        if (description != other.description) return false
        if (plannedDuration != other.plannedDuration) return false

        return true
    }

    override fun hashCode(): Int {
        var result = description.hashCode()
        result = 31 * result + plannedDuration.hashCode()
        return result
    }

    override fun toString(): String {
        return "Leg(description='$description', plannedDuration=$plannedDuration)"
    }


}